packages/vscode-extension/src/connection.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  Neo4jDataSummary,
+  testConnectionTransiently,
+} from '@neo4j-cypher/schema-poller';
+import { getContext } from './appContext';
+import {
+  MethodName,
+  sendNotificationToLanguageClient,
+} from './languageClientService';
+
+export type Connection = {
+  key: string;
+  name: string;
+  scheme: string;
+  host: string;
+  port: string;
+  user: string;
+  database: string;
+  connect: boolean;
+  default: boolean;
+  dataSummary: Neo4jDataSummary;
+};
+
+type Connections = {
+  [key: string]: Connection | null;
+};
+
+const CONNECTIONS_KEY: string = 'connections';
+
+export async function testConnection(
+  connection: Connection,
+  password: string,
+): Promise<boolean> {
+  return await testConnectionTransiently(
```

Comment:
I wonder if there's a better way to do this.. this function creates a driver, checks that the credentials are valid and there's connectivity, then disposes the driver.. this seems wasteful, but I can't think of a better way..